### ISSUE TITLE
Performance drop after >19.03.1 updates

### ISSUE THREAD

**Issue by markomitranic** [boot2docker/boot2docker#1405]
Hey guys and girls,

i've had a weird disk IO performance drop with bound volumes since update to anything larger than 19.03.01.

I have successfully replicated the issue numerous times, on at least 5 comupters in my team. We all use macos catalinas or mojaves, with fully updated virtualboxes. I have mitigated the issue by forcing everyone to use 19.03.01, but am really interested in what exact change could have caused this drop? Especially since i went through the changelog, and saw nothing in particular...

If anyone has any details that might help me in researching further, please... :)

**Comment by tianon** [comment]
I'd have to guess something in the kernel change from 4.14.134 to 4.14.141, but if you're doing IO on the shared folders from VirtualBox, it could very easily be something in those VirtualBox changes (which would be a shame, because we have to drop support for VirtualBox 5 entirely to go with 6.x guest additions, which is why that revert is in there).

**Comment by tianon** [comment]
Regarding building locally, I just do this:

```console
$ docker build -t boot2docker/boot2docker --pull --force-rm .
$ docker run --rm boot2docker/boot2docker > boot2docker.iso
```

**Comment by markomitranic** [comment]
@tianon i doubt it, im using the same 6.0.x version of vbox on all machines and b2d versions. Thanks for the info, i’ll do a couple of builds and  see if i can pinpoint the exact change that replicates the issue. and report back ofc.

the newer versions are unfortunately unusable. the pageload performance of symfony is on par with docker for mac if not worse. (which means real real bad. like 40 sec bad)

**Comment by markomitranic** [comment]
> ```
> docker run --rm boot2docker/boot2docker > boot2docker.iso
> ```

Hmm, this does not seem to work, a clean checkout of 19.03.1 does not build properly:


`+ for mirror in $TCL_MIRRORS
+ wget -O usr/local/tce.installed/optional/lzo.tcz http://repo.tinycorelinux.net/10.x/x86_64/tcz/lzo.tcz
--2019-11-27 09:33:09--  http://repo.tinycorelinux.net/10.x/x86_64/tcz/lzo.tcz
Resolving repo.tinycorelinux.net (repo.tinycorelinux.net)... 89.22.99.37
Connecting to repo.tinycorelinux.net (repo.tinycorelinux.net)|89.22.99.37|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 57344 (56K) [text/plain]
Saving to: 'usr/local/tce.installed/optional/lzo.tcz'

     0K                                    100% 18.2M=0.003s

2019-11-27 09:33:09 (18.2 MB/s) - 'usr/local/tce.installed/optional/lzo.tcz' saved [57344/57344]

+ cd usr/local/tce.installed/optional
+ md5sum -c lzo.tcz.md5.txt
lzo.tcz: FAILED
md5sum: WARNING: 1 computed checksum did NOT match
+ cd usr/local/tce.installed/optional
+ md5sum -c lzo.tcz.md5.txt
lzo.tcz: FAILED
md5sum: WARNING: 1 computed checksum did NOT match
Removing intermediate container 8d357f2f1737
The command '/bin/bash -Eeuo pipefail -xc for package in $TCL_PACKAGES; do 		eval 'md5="$TCL_PACKAGE_MD5__'"$(echo "$package" | sed -r 's/[^a-zA-Z0-9]+/_/g')"'"'; 		echo "$md5 *$package" > "usr/local/tce.installed/optional/$package.md5.txt"; 		for mirror in $TCL_MIRRORS; do 			if 			wget -O "usr/local/tce.installed/optional/$package" "$mirror/$TCL_MAJOR/x86_64/tcz/$package" 				&& ( cd usr/local/tce.installed/optional && md5sum -c "$package.md5.txt" ) 			; then 			break; 			fi; 		done; 		( cd usr/local/tce.installed/optional && md5sum -c "$package.md5.txt" ); 		unsquashfs -dest . -force "usr/local/tce.installed/optional/$package"; 		touch "usr/local/tce.installed/${package%.tcz}"; 		touch "usr/local/tce.installed/optional/$package.dep"; 	done; 		tcl-chroot ldconfig; 	for script in usr/local/tce.installed/*; do 		[ -f "$script" ] || continue; 		[ -x "$script" ] || continue; 		tcl-chroot "$script"; 	done; 		{ 		echo '#!/bin/bash -Eeux'; 		echo 'tcl-chroot su -c "tce-load -wicl \"\$@\"" docker -- - "$@"'; 	} > /usr/local/bin/tcl-tce-load; 	chmod +x /usr/local/bin/tcl-tce-load' returned a non-zero code: 1
`

**Comment by tianon** [comment]
Correct, that older tag will not build as-is because the TCL artifacts have been updated.  I'd recommend instead tweaking the knobs (`LINUX_ VERSION`, etc) on the tip of the 19.03.x branch instead (or playing with `./update.sh` which is what I use to update those checksums).

**Comment by sspat** [comment]
Same issue here, sadly found this issue only after spending hours reinstalling everything.

**Comment by markomitranic** [comment]
Thank you for the reply @sspat I am very sory that i havent done what you did, but i simply had no time in the previous months. What i did do with my testing is very similar to results you've got. All of my 30 developers still use 19.03.1 and we are stuck with it for good it seems, especially since Docker itself would be discontinuing the docker-machine project.

We will literally have to spin up a CND cluster or something similar. Which is a complete overkill for this kind of development.

I would hope @tianon has some idea of what we may have missed?
